clients/src/main/java/org/apache/kafka/clients/consumer/KafkaConsumer.java

```diff
@@ -1535,6 +1561,7 @@ public void seekToBeginning(Collection<TopicPartition> partitions) {
             for (TopicPartition tp : parts) {
                 log.debug("Seeking to beginning of partition {}", tp);
                 subscriptions.requestOffsetReset(tp, OffsetResetStrategy.EARLIEST);
+                metadata.resetLastSeenEpoch(tp);
```

Comment:
This should probably go before the above line since that offset reset might update the metadata